ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -205,7 +205,8 @@ private KsqlEntity executeStatement(
             || statement instanceof DropStream
             || statement instanceof DropTable
     ) {
-      //   getStatementExecutionPlan(statement, statementText, streamsProperties);
+      //Sanity check for the statement before distributing it.
+      getStatementExecutionPlan(statement, statementText, streamsProperties);
```

Comment:
I think this should be named `validateExecutionPlan` so its intended use is clear. The comment only helps people reading code in this context, but this method could be used in other contexts. 

If this method is doing two things (ie. generating and validating the plan), then it would be better to split it up into two separate methods, each performing a particular function.

Reply:
I agree w @apurvam  - it should be renamed. Looking the code it doesn't make - sense. It looks like it is executing various ddlcommands inside getStatementExecutionPlan() (i.e. CreateTableAsSelect etc) - and also publishes them onto the command topic to allow all nodes to execute the same command again  "distributeStatement()". Shouldnt we have a validate() - that doesn't do any execution and then call distribute()

Reply:
Good point. I added a new method and updated the PR.